sdk/identity/azure-identity/azure/identity/_credentials/azure_cli.py

```diff
@@ -35,14 +36,11 @@ class AzureCliCredential(object):
     """Authenticates by requesting a token from the Azure CLI.
 
     This requires previously logging in to Azure via "az login", and will use the CLI's currently logged in identity.
-
-    :keyword bool allow_multitenant_authentication: when True, enables the credential to acquire tokens from any tenant
-        the identity logged in to the Azure CLI is registered in. When False, which is the default, the credential will
-        acquire tokens only from the tenant of the Azure CLI's active subscription.
     """
 
     def __init__(self, **kwargs):
-        self._allow_multitenant = kwargs.get("allow_multitenant_authentication", False)
+        disable_multitenant = os.environ.get(EnvironmentVariables.AZURE_IDENTITY_DISABLE_MULTITENANTAUTH, False)
```

Comment:
Suggest centralizing this in `resolve_tenant` instead

Reply:
The problem is we want to handle 

```py
if self._tenant_id == "adfs":
            self._allow_multitenant = False
```

I don't want to move it into resolve_tenant() as well because self._tenant_id is not always accessible from resolve_tenant().

Any suggestions?

Reply:
When isn't it accessible? resolve_tenant's first argument is the calling credential's configured tenant.